### PR TITLE
Issue #11709 fix jetty-with-custom-class.xml file and use it

### LINKED ISSUE #11709: test-jetty-ee9-osgi file jetty-with-custom-class.xml references an ee10 class
**jetty-12.0.x**

**ee9**

### PR DESCRIPTION
**Pull request by janbartel**
Fixes #11709

Also fix:

+ jetty osgi logging
+ ensure DeploymentManager is configured
+ re-instate test that uses jetty-with-custom-class.xml file

### CODE REVIEW

**Comment by janbartel** [comment]
@olamy bump

File: jetty-core/jetty-osgi/src/main/java/org/eclipse/jetty/osgi/JettyServerFactory.java [hunk 1]
```diff
@@ -195,13 +196,16 @@ public static Server createServer(String name, Dictionary<String, Object> props,
    private static DeploymentManager ensureDeploymentManager(Server server)
    {
        Collection<DeploymentManager> deployers = server.getBeans(DeploymentManager.class);
-       DeploymentManager deploymentManager;
+       DeploymentManager deploymentManager = null;
 
        if (deployers != null)
        {
-           deploymentManager = deployers.stream().findFirst().get();
+           Optional<DeploymentManager> opt = deployers.stream().findFirst();
```
Comment (olamy) [inline jetty-core/jetty-osgi/src/main/java/org/eclipse/jetty/osgi/JettyServerFactory.java]: 3 lines could be a single line :) 
```
deploymentManager = deployers.stream().findFirst().orElse(null);
```

File: jetty-ee10/jetty-ee10-osgi/test-jetty-ee10-osgi/pom.xml [hunk 2]
```diff
@@ -42,6 +42,18 @@
         </exclusion>
       </exclusions>
     </dependency>
+    <!--
```
Comment (olamy) [inline jetty-ee10/jetty-ee10-osgi/test-jetty-ee10-osgi/pom.xml]: just remove commented section

File: jetty-ee10/jetty-ee10-osgi/test-jetty-ee10-osgi/pom.xml [hunk 3]
```diff
@@ -42,6 +42,18 @@
...
+    <dependency>
+       <groupId>org.slf4j</groupId>
+       <artifactId>slf4j-simple</artifactId>
+       <scope>test</scope>
+     </dependency>
+     -->
+    <dependency>
+      <groupId>org.eclipse.jetty</groupId>
+      <artifactId>jetty-slf4j-impl</artifactId>
+      <!-- scope>test</scope -->
```
Comment (olamy) [inline jetty-ee10/jetty-ee10-osgi/test-jetty-ee10-osgi/pom.xml]: remove comment

**Review by olamy — DISMISSED** [review]
LGTM (just some nit comments)

**Comment by janbartel** [comment]
@olamy suggested changes made - can you give me another review so I can merge??

**Review by olamy — APPROVED** [review]

**Pull request merged**